migration_test.go

```diff
@@ -132,6 +132,49 @@ func Test0001MigrationStepUp(t *testing.T) {
 	checkAllExpectations(t, mock)
 }
 
+// Test0001MigrationStepUpOnMigrationFailure test checks migration #1 in case
+// the table create fails.
+func Test0001MigrationStepUpOnMigrationFailure(t *testing.T) {
+	// error to be thrown
+	mockedError := errors.New("mocked error")
+
+	// prepare new mocked connection to database
+	connection, mock := mustCreateMockConnection(t)
+
+	// prepare mocked result for SQL query
+	// initial database version
+	rows := sqlmock.NewRows([]string{"version"})
+	rows.AddRow("0")
+
+	count := sqlmock.NewRows([]string{"count"})
+	count.AddRow("1")
+
+	// expected query performed by tested function
+	expectedQuery0 := "SELECT COUNT\\(\\*\\) FROM migration_info;"
+	expectedQuery1 := "SELECT version FROM migration_info;"
+	expectedCreate := "CREATE TABLE IF NOT EXISTS event_targets .*"
+
+	// queries to retrieve DB version should succeed
+	mock.ExpectQuery(expectedQuery0).WillReturnRows(count)
+	mock.ExpectQuery(expectedQuery1).WillReturnRows(rows)
+	mock.ExpectBegin()
+
+	// table create will fail
+	mock.ExpectExec(expectedCreate).WillReturnError(mockedError)
+
+	// so we expect roll back instead of transaction commit
+	mock.ExpectRollback()
+	mock.ExpectClose()
+
+	utils.Set(main.All())
+
+	// migrate to version 1
+	assert.Error(t, main.Migrate(connection, 1), mockedError)
```

Comment:
I believe that `mockedError` is not needed, that 3rd argument is to send a message in case of test failure, something like `"expected error but got nil"`   or we could
`mockedError := errors.New("test failed: expected error")`